lua/ruby_nvim/util.lua

```diff
@@ -101,4 +117,51 @@ M.alternate = function(file_path)
   end
 end
 
+M.all_methods = function ()
+  local bufnr = vim.api.nvim_get_current_buf()
+  local ft = vim.api.nvim_buf_get_option(bufnr, "ft")
+  local tree = vim.treesitter.get_parser(bufnr, ft):parse()[1]
+  local root = tree:root()
+  return get_methods(root)
+end
+
+M.get_gem_name = function (node, bufnr)
+  while node ~= nil do
+    if node:type() == 'call' then
+      local method = node:field('method')[1]
+      local method_name = vim.treesitter.query.get_node_text(method, bufnr)
+      if method_name == 'gem' then
+        local method_args = node:field('arguments')[1] --that when using get_node_text has text 'aws-sdk-s3', '~> 1'
+        local gem_node_with_quotes = method_args:child(0) -- that when using get_node_text has text 'aws-sdk-s3'
+        local gem_node = gem_node_with_quotes:child(1) -- that when usin get_node_text has text aws-sdk-s3
+        local gem_name = vim.treesitter.query.get_node_text(gem_node, bufnr)
+        if gem_name ~= "'" then
```

Comment:
Can we avoid having a single quote as a gem name? When does this happen?

Reply:
When on gemfile you set
```ruby
gem ''
```

Reply:
Wow… bizarre. I mean: is it useful? Why should we expect `gem ''` in a `Gemfile`?
Anyway. Good to go, I guess.